tests/integration.rs

```diff
@@ -4855,12 +4856,30 @@ mod common_parallel {
             .args(["--memory", format!("size={guest_memory_size_kb}K").as_str()])
             .args(["--kernel", kernel_path.to_str().unwrap()])
             .args(["--cmdline", DIRECT_KERNEL_BOOT_CMDLINE])
+            .args(["--event-monitor", format!("path={event_path}").as_str()])
             .default_disks()
             .capture_output()
             .spawn()
             .unwrap();
 
-        thread::sleep(std::time::Duration::new(20, 0));
```

Comment:
Shouldn't this just be replaced by `guest.wait_vm_boot(None)` etc?

Reply:
I time tested test_memory_overhead

1. With guest.wait_vm_boot(None);
DEFAULT_TCP_LISTENER_TIMEOUT: i32 = 120
real    2m15.147s
user    0m43.166s
sys     0m17.518s

2. As it is now on the main branch
thread::sleep(std::time::Duration::new(20, 0));
real    0m32.502s
user    0m40.003s
sys     0m15.313s

3. With event monitor - as it is on this PR
real    0m12.288s
user    0m28.119s
sys     0m15.694s

Evidently event monitor is the fastest.

Reply:
I know it's confusing but the VMM concept of "booted" isn't really related to the definition of the guest boot. All it means is that the vCPU threads have been started and are running in the event loop. It doesn't say anything about the guest OS state whereas `guest.vm_wait_boot()` actually waits for the guest to boot and relies on a TCP connection from the guest.

Reply:
Created #5738 to track that work - might be a good idea to do all those first before focussing on the event monitor.